### PR TITLE
Default to rv64gc on riscv64

### LINKED ISSUE #4375: core/thread/osthread.d(1594):1:2: error: instruction requires the following: 'D' (Double-Precision Floating-Point)
Building ldc fails on riscv64:

core/thread/osthread.d(1594):1:2: error: instruction requires the following: 'D' (Double-Precision Floating-Point)
        fsd fs0, 0(a2)
        ^
core/thread/osthread.d(1594):1:2: error: instruction requires the following: 'D' (Double-Precision Floating-Point)
        fsd fs1, 0(a2)
        ^
core/thread/osthread.d(1594):1:2: error: instruction requires the following: 'D' (Double-Precision Floating-Point)
        fsd fs2, 0(a2)
        ^
core/thread/osthread.d(1594):1:2: error: instruction requires the following: 'D' (Double-Precision Floating-Point)
        fsd fs3, 0(a2)
        ^
core/thread/osthread.d(1594):1:2: error: instruction requires the following: 'D' (Double-Precision Floating-Point)
        fsd fs4, 0(a2)
        ^
core/thread/osthread.d(1594):1:2: error: instruction requires the following: 'D' (Double-Precision Floating-Point)
        fsd fs5, 0(a2)
        ^
core/thread/osthread.d(1594):1:2: error: instruction requires the following: 'D' (Double-Precision Floating-Point)
        fsd fs6, 0(a2)
        ^
core/thread/osthread.d(1594):1:2: error: instruction requires the following: 'D' (Double-Precision Floating-Point)
        fsd fs7, 0(a2)
        ^
core/thread/osthread.d(1594):1:2: error: instruction requires the following: 'D' (Double-Precision Floating-Point)
        fsd fs8, 0(a2)
        ^
core/thread/osthread.d(1594):1:2: error: instruction requires the following: 'D' (Double-Precision Floating-Point)
        fsd fs9, 0(a2)
        ^
core/thread/osthread.d(1594):1:2: error: instruction requires the following: 'D' (Double-Precision Floating-Point)
        fsd fs10, 0(a2)
        ^
core/thread/osthread.d(1594):1:2: error: instruction requires the following: 'D' (Double-Precision Floating-Point)
        fsd fs11, 0(a2)
        ^
core/thread/osthread.d(1594):1:2: error: instruction requires the following: 'D' (Double-Precision Floating-Point)
        fsd fs0, 0(a0)
        ^
core/thread/osthread.d(1594):1:2: error: instruction requires the following: 'D' (Double-Precision Floating-Point)
        fsd fs1, 0(a0)
        ^
core/thread/osthread.d(1594):1:2: error: instruction requires the following: 'D' (Double-Precision Floating-Point)
        fsd fs2, 0(a0)
        ^
core/thread/osthread.d(1594):1:2: error: instruction requires the following: 'D' (Double-Precision Floating-Point)
        fsd fs3, 0(a0)
        ^
core/thread/osthread.d(1594):1:2: error: instruction requires the following: 'D' (Double-Precision Floating-Point)
        fsd fs4, 0(a0)
        ^
core/thread/osthread.d(1594):1:2: error: instruction requires the following: 'D' (Double-Precision Floating-Point)
        fsd fs5, 0(a0)
        ^
core/thread/osthread.d(1594):1:2: error: instruction requires the following: 'D' (Double-Precision Floating-Point)
        fsd fs6, 0(a0)
        ^
core/thread/osthread.d(1594):1:2: error: instruction requires the following: 'D' (Double-Precision Floating-Point)
        fsd fs7, 0(a0)
        ^
core/thread/osthread.d(1594):1:2: error: instruction requires the following: 'D' (Double-Precision Floating-Point)
        fsd fs8, 0(a0)
        ^
core/thread/osthread.d(1594):1:2: error: instruction requires the following: 'D' (Double-Precision Floating-Point)
        fsd fs9, 0(a0)
        ^
core/thread/osthread.d(1594):1:2: error: instruction requires the following: 'D' (Double-Precision Floating-Point)
        fsd fs10, 0(a0)
        ^
core/thread/osthread.d(1594):1:2: error: instruction requires the following: 'D' (Double-Precision Floating-Point)
        fsd fs11, 0(a0)
        ^
make[2]: *** [runtime/CMakeFiles/druntime-ldc-shared.dir/build.make:346: runtime/objects-shared/core/atomic.o] Error 1
make[2]: Leaving directory '/home/abuild/rpmbuild/BUILD/ldc-1.32.0-src/build-bootstrap'
make[1]: *** [CMakeFiles/Makefile2:493: runtime/CMakeFiles/druntime-ldc-shared.dir/all] Error 2
make: *** [Makefile:169: all] Error 2

### PR DESCRIPTION
**Pull request by andreas-schwab**
Fixes #4375

### CODE REVIEW

**Comment by JohanEngelen** [comment]
I have no opinion/expertise on this, but we shouldn't be switching back and forth between defaults every release.

@zyedidia  @andreas-schwab 
Can you please converge on a solution of the discussions about this topic? (e.g. https://github.com/ldc-developers/ldc/pull/4322)

Please add to this PR a changelog/release notes entry about this.

Possibly related: https://github.com/llvm/llvm-project/issues/61991

**Comment by kinke** [comment]
No strong opinion from my side either, but I think the default should account for all features the D language mandates, e.g., double-precision floating point support etc.

**Comment by kalev** [comment]
Fedora's riscv64 appears to be built using --with-arch=rv64gc --with-abi=lp64d (taken from gcc configure invocation) so from my point of view, this PR seems like a step in the right direction.

**Comment by zyedidia** [comment]
As long as we have options to change the enabled extensions and abi (which we do as far as I know) then I think this default is reasonable and consistent with other toolchains. Ideally I think we would have the `-march` flag support specifying selected ISA directly (like `rv64imac`) rather than needing to use `mattr` to disable undesired extensions included in the default.

**Comment by andreas-schwab** [comment]
Made a silly typo, retesting.

File: driver/targetmachine.cpp [hunk 1]
```diff
@@ -120,6 +127,10 @@ const char *getABI(const llvm::Triple &triple) {
   case llvm::Triple::ppc64le:
     return "elfv2";
   case llvm::Triple::riscv64:
+    if (hasFeature("d"))
+      return "lp64d";
```
Comment (kinke) [inline driver/targetmachine.cpp:131]: So a `-mattr=-d` would get me a `lp64d` ABI; is that intended? Or should it be narrowed to a `+d`/`+f` only?

An entry in `CHANGELOG.md` would be nice too. :)

**Comment by kinke** [comment]
Thx Andreas!

**Pull request merged**